opentech/apply/funds/views.py

```diff
@@ -222,3 +227,104 @@ def form_valid(self, form):
         self.object.form_data = form.cleaned_data
         self.object.save()
         return HttpResponseRedirect(self.get_success_url())
+
+
+class RevisionListView(TemplateView):
+    template_name = 'funds/revisions_list.html'
+
+    def get_context_data(self, **kwargs):
+        submission = ApplicationSubmission.objects.get(id=self.kwargs['submission_pk'])
+        revisions = [{
+            'date': "2018/06/14",
+            'author': User.objects.first(),
+            'id': submission.id,
+        }]
+        revisions.extend([{
+            'date': "2018/06/14",
+            'author': User.objects.all()[i],
+            'id': ApplicationSubmission.objects.order_by('?').first().id,
+        } for i in range(5)])
+        return super().get_context_data(
+            submission=submission,
+            revisions=revisions,
+            **kwargs,
+        )
+
+
+class RevisionCompareView(TemplateView):
+    template_name = 'funds/revisions_compare.html'
+
+    def wrap(self, class_name, text):
+        return format_html('<span class="{}">{}</span>', class_name, mark_safe(text))
+
+    def deleted(self, text):
+        return self.wrap('deleted', text)
+
+    def added(self, text):
+        return self.wrap('added', text)
+
+    def compare_answer(self, answer_a, answer_b):
+        if not answer_a and not answer_b:
+            # This catches the case where both results are None and we cant compare
+            return answer_b
+        if isinstance(answer_a, dict) or isinstance(answer_b, dict):
+            # TODO: handle file dictionaries
+            return answer_b
+
+        diff = SequenceMatcher(None, answer_a, answer_b)
+        output = []
+        added = ''
+        deleted = ''
+        for opcode, a0, a1, b0, b1 in diff.get_opcodes():
+            if opcode == 'equal':
+                if a1 - a0 > 2 or not (added or deleted):
```

Comment:
Is this the "if diff is more than 2 chars, show the word diff, otherwise do the letter diff inline" logic?
Do document!

Reply:
Will improve documentation. thanks dan.

Reply:
`s/Do document!/add a comment about this/`